src/keepkey.ts

```
@@ -172,11 +172,21 @@ export default class KeepKey {
   public async ethereumGetAddress (a: Messages.EthereumGetAddress.AsObject): Promise<string> {
     const getAddr = new Messages.EthereumGetAddress()
     getAddr.setAddressNList(a.addressNList)
-    getAddr.setShowDisplay(a.showDisplay)
+    getAddr.setShowDisplay(a.showDisplay !== false)
     // send, receive ethereumaddress message
     const [_, response] = await this.device.exchange(Messages.MessageType.MESSAGETYPE_ETHEREUMGETADDRESS, getAddr)
-    const { address, addressStr } = (response as Messages.EthereumAddress).toObject()
-    return addressStr || ('0x' + (address as string))
+    const ethAddress = response as Messages.EthereumAddress
+
+    let result
+    if (ethAddress.hasAddressStr()) {
+      result = ethAddress.getAddressStr()
+    } else if (ethAddress.hasAddress()) {
+      result = '0x' + toHexString(ethAddress.getAddress_asU8())
+    } else {
+      console.log(`unable to obtain eth address from device`)
```

Should we `throw` here if this is an unexpected error?

---

yeah, that'd be prudent

---

maybe, wasn't sure the best approach. throw instead?

---

throw + early return

---

prefer multiple return statements? ok

---

yeah. control flow is easier to read than dataflow

---

s/read/reason about/

---

/g